File: app/src/main/java/com/gmail/jorgegilcavazos/ballislife/features/games/GamesView.java
package com.gmail.jorgegilcavazos.ballislife.features.games;

import com.gmail.jorgegilcavazos.ballislife.features.model.GameV2;

import java.util.Calendar;
import java.util.List;

public interface GamesView {

    void setLoadingIndicator(boolean active);

    void setDateNavigatorText(String dateText);

    void hideGames();

    void showGames(List<GameV2> games);

    void showGameDetails(GameV2 game, Calendar selectedDate);

    void updateScores(List<GameV2> games);

    void setNoGamesIndicator(boolean active);

    void showNoNetSnackbar();

    void showErrorSnackbar();

    void dismissSnackbar();
}

File: app/src/main/java/com/gmail/jorgegilcavazos/ballislife/features/games/GamesView.kt
package com.gmail.jorgegilcavazos.ballislife.features.games

import com.gmail.jorgegilcavazos.ballislife.features.games.GamesUiEvent.*
import com.gmail.jorgegilcavazos.ballislife.features.model.GameV2
import io.reactivex.Observable
import java.util.*

interface GamesView {

  fun openGameEvents(): Observable<OpenGameEvent>

  fun dateSelectionEvents(): Observable<DateSelectedEvent>

  fun loadGamesEvents(): Observable<LoadGamesEvent>

  fun refreshGamesEvents(): Observable<RefreshGamesEvent>

  fun setLoadingIndicator(active: Boolean)

  fun setDateNavigatorText()

  fun hideGames()

  fun showGames(games: List<GameV2>)

  fun showGameDetails(game: GameV2)

  fun setNoGamesIndicator(active: Boolean)

  fun showNoNetSnackbar()

  fun showErrorSnackbar(code: Int)

  fun dismissSnackbar()

  fun getCurrentDateShown(): Calendar
}
